lib/rubocop/cop/performance/double_start_end_with.rb

```diff
@@ -0,0 +1,47 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop checks for double `#start_with?` or `#end_with?` calls
+      # separated by `||`. In some cases such calls can be replaced
+      # with an single `#start_with?`/`#end_with?` call.
+      #
+      # @example
+      #
+      #   @bad
+      #   str.start_with?("a") || str.start_with?(Some::CONST)
+      #   str.start_with?("a", "b") || str.start_with?("c")
+      #   var1 = ...
+      #   var2 = ...
+      #   str.end_with?(var1) || str.end_with?(var2)
+      #
+      #   @good
+      #   str.start_with?("a", Some::CONST)
+      #   str.start_with?("a", "b", "c")
+      #   var1 = ...
+      #   var2 = ...
+      #   str.end_with?(var1, var2)
+      class DoubleStartEndWith < Cop
+        MSG = 'Use `str.%{method}(x, ..., y, ...)` ' \
+              'instead of `str.%{method}(x, ...) || str.%{method}(y, ...)`.'
```

Comment:
Rather than showing `x, ...` and `y, ...`, it would be nicer if the message showed the *actual* arguments.